Add comment to product

Jorik88 opened the pull request:
I suggest adding an opportunity to add comments to the products.

Jorik88 committed:
created new table for added comments on product

Jorik88 committed:
created new entity ProductComment for mapping with table product_comment
added constants for change status of comment

Jorik88 committed:
corrected the code according to the requirements of the collaborator

Jorik88 committed:
added new version of spring and hibernate
made code review for new dependency
added new dao and some implementation
added tests for dao
added builders for tests

Jorik88 committed:
in implementation ProductCommentDao switched to use JPA
added javadoc for dao
added integration tests for dao
rename column name in entity by code convention

lelik1979 commented on beads_model/src/main/java/com/beads/model/dao/ProductCommentDaoImpl.java:
```diff
@@ -0,0 +1,85 @@
+@Repository
+public class ProductCommentDaoImpl extends BaseDao implements ProductCommentDao {
+
+  /**
```
java doc better to put on interface.

lelik1979 commented on beads_model/src/main/java/com/beads/model/dao/ProductCommentDaoImpl.java:
```diff
@@ -0,0 +1,85 @@
+   * @return loaded comment
+   */
+  @Override
+  public ProductComment loadCommentById(Integer id) {
```
better rename it to `getCommentById`
if you use hibernate `get` because hibernate has `load` method

lelik1979 commented on beads_model/src/main/java/com/beads/model/domain/ProductComment.java:
```diff
@@ -0,0 +1,143 @@
+
+  @Enumerated(EnumType.STRING)
+  @Column(name = "status", columnDefinition = "ENUM('VISIBLE', 'INVISIBLE')")
+  private CommentStatus status = CommentStatus.INVISIBLE;
```
better group constant together for readability.

lelik1979 commented on beads_model/src/test/java/com/beads/model/dao/ProductDaoImplTest.java:
```diff
@@ -31,4 +33,12 @@ public void testLoadProductsByGroupId() throws Exception {
         Product pr =productDao.loadProductById(1);
     }
+
+//    @Rollback(false)
```
better not to check-in commented code

lelik1979 commented on beads_model/src/test/java/com/beads/model/dao/ProductDaoImplTest.java:
```diff
@@ -31,4 +33,12 @@ public void testLoadProductsByGroupId() throws Exception {
+
+//    @Rollback(false)
+    @Test
+    public void testSaveProduct() {
```
what does this test check ?
do ever need it ?

Jorik88 committed:
transferred javadoc in interface
fixed test for ProductDaoImpl
rename method loadCommentById in ProductCommentDaoImpl to getCommentById

lelik1979 reviewed (approved):
looks good to merge
